plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -343,10 +356,23 @@ public function addDocument($data, $type)
         unset($data['id']);
 
         $document = new \Elastica\Document($id, $data);
-        $document->setType($type);
+
+        // Setting a dummy type since it is required in ES 6.x
+        // but it can be removed in 7.x when it becomes optional
+        $document->setType(self::ES_TYPE);
+
+        if (!$this->currentBatchType) {
+            $this->currentBatchType = $type;
+        }
 
         if ($this->batchMode) {
             // Add this document to the batch add queue
+            if ($this->currentBatchType != $type) {
+                $this->flushBatch();
+                $this->currentBatchType = $type;
+                $this->index->refresh();
```

Comment:
Sorry, I meant refresh when I said flush. This is my suggestion to avoid refreshing all indices when we only flush the batch in one of them:

* Looking at the calls of `$this->index->refresh();` they all happen after `$this->flushBatch();`. So I'd move that call inside the flushBatch method and remove it from all other places.
* Now that you are sure it only happens there, you also know that we only flush the batch for one index, so you could change the call to include the index name: `$this->index->refresh($this->currentBatchIndexName);`.
* And the refresh function in the wrapper could require a `$name` parameter and only refresh that index:
  
  ```php
  public function refresh($name)
  {
      $this->indices[$name]->refresh();
  }
  ```

Additionally, similar to the comment I made about the update methods, up to you if you want to include it in the wrapper or not, you could do it outside like you are doing now for the updates: `$this->index->getIndex($this->currentBatchIndexName)->refresh();`. I think it's important to be consistent so, now that we are going with the wrapper, I think it makes more sense to put everything there.

Reply:
I think what I suggested above:

> Looking at the calls of $this->index->refresh(); they all happen after $this->flushBatch();. So I'd move that call inside the flushBatch method and remove it from all other places.

It would have simplified the latest changes and the code in general, only one call to refresh inside flushBatch, and always using `$this->currentBatchIndexName`, no need to make the index name optional since it always happens per index.

Sorry if that wasn't clear, and feel free to leave it as it is ;)